effekt/shared/src/main/scala/effekt/machine/Transformer.scala

```diff
@@ -133,24 +140,34 @@ object Transformer {
           Clause(List(transform(lifted.ValueParam(id, binding.tpe))), transform(rest)),
             transform(binding)
         )
+
+      // TODO deal with BlockLit
       case lifted.App(lifted.BlockVar(id, tpe), targs, args) =>
         if(targs.exists(requiresBoxing)){ ErrorReporter.abort(s"Types ${targs} are used as type parameters but would require boxing.") }
-        // TODO deal with BlockLit
-        id match {
-          case symbols.BlockParam(_, _) =>
+
+        BPC.info(id) match {
```

Comment:
Instead of inspecting symbols, we now dispatch on what we know (that is, recorded) about the binder of the variable.